source/computer/macos-setup.rst

```diff
@@ -1,9 +1,159 @@
 macOS 配置指南
 ==============
 
+:本节作者: `王亮 <https://github.com/wangliang1989>`__\ （作者）、
+           `田冬冬 <https://me.seisman.info/>`__\ （作者）、
+           `姚家园 <https://github.com/core-man>`__\ （审稿）
+:最近更新日期: 2021-01-11
+
+----
+
+.. warning::
+
+   本文成稿于 2018 年 1 月，并于 2021 年 1 月做少量改动。
+   但笔者尚未在新电脑上进行测试，无法保证所有命令均正确。欢迎反馈。
+
+第一次启动 Mac 电脑后，经过简单的设置，就得到了一个可供日常使用的 macOS 系统，
+但这样的 macOS 系统尚不能满足日常科研与编程开发的需求，还需要做进一步的配置。
+
+Command Line Tools for Xcode
+----------------------------
+
+Xcode 是 macOS 下的集成开发环境（IDE），类似于 Windows 下的 Microsoft Visual Studio。
+Command Line Tools for Xcode 是 Xcode 的一个子集，其包含了常用的命令行开发工具，
+比如 C 编译器、make、git 等，是安装很多其他软件的基础。
+
+按下 :kbd:`Command+空格`\ ，搜索并打开“Terminal”。在 Terminal 中执行如下命令
+以安装 Command Line Tools for Xcode::
+
+   $ xcode-select --install
+
+在弹出的窗口点击“Install”即可。
+
+此处安装的 Command Line Tools 可能不是最新版，可以到“App Store”的“更新”中或
+“设置”的“软件更新”中查看是否有相关更新。如果有，则升级到最新版。
+
+Homebrew
+--------
+
+`Homebrew <https://brew.sh/index_zh-cn.html>`__ 是 macOS 下的第三方软件包管理器。
+可以用于安装各种常见的软件包、库文件以及字体。日常及科研工作中所需的大多数软件、
+库文件以及字体都可以通过 Homebrew 安装。
+
+要安装 Homebrew，需要将如下命令复制到终端中并执行::
+
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

Comment:
When I install Homebrew, it seems CLT for Xcode is also installed.

```
...
==> Installing Command Line Tools for Xcode-12.3
==> /usr/bin/sudo /usr/sbin/softwareupdate -i Command\ Line\ Tools\ for\ Xcode-12.3
Software Update Tool

Finding available software

Downloading Command Line Tools for Xcode
Downloaded Command Line Tools for Xcode
Installing Command Line Tools for Xcode

Done with Command Line Tools for Xcode
Done.
...
```

Reply:
It seems homebrew will install CLT if not already installed.

Reply:
Did you install CLT before homebrew?

Reply:
If it is installed, it seems it tries to install or update it?

Reply:
I think so. It does no harm to install CLT and homebrew separately.

Reply:
In this way, which gcc will be used?
```
$ which gcc
/usr/bin/gcc
```

Reply:
`/usr/bin/gcc` is from CLT, and it's not real GNU GCC, it's Apple Clang, but it's called gcc for compatibility.

Reply:
All my C codes are compiled using this gcc (i.e., Apple Clang), and it works well to me.

Reply:
Will CLT install gcc in this folder too?

Reply:
`/usr/bin/gcc` is installed by CLT.

Reply:
I don't where homebrew will install gcc.

Reply:
Homebrew installs everything to `/usr/local`. The homebrew gcc is only used by homebrew itself, so you won't be able to use the homebrew gcc (the real GNU GCC) unless you manually add the bin path to PATH.

Reply:
Actually, I cannot find any gcc in `/usr/local`

Reply:
Did you run `brew install gfortran`?

Can you find `/usr/local/gcc-10`?